tests/Fixer/FunctionNotation/FunctionDeclarationFixerTest.php

```diff
@@ -161,6 +171,141 @@ function foo() /* bar */
      * @param string      $expected
      * @param null|string $input
      *
+     * @dataProvider provideClosureFunctionSpacingNoneCases
+     */
+    public function testFixClosureFunctionSpacingNone($expected, $input = null)
```

Comment:
we use config as 3rd param ;)
I don't see a need for `casesForConfigurations` method or merging arrays via `+`.

just

```php
function provide...() {
    return [
        [ $expected, $input, ],
        [ $expected, $input, $config, ],
    ];
}
```